BaseCollector.py

```diff
@@ -1,11 +1,64 @@
+import json
 from abc import ABC, abstractmethod
+from ucsmsdk.ucsexception import UcsException
+from modules.UcsmServer import UcsmServer
 
 
 class BaseCollector(ABC):
+    def __init__(self, creds, inventory_file):
+        self.creds = creds
+        self.inventory_file = inventory_file
+        self.handles = {}
+
     @abstractmethod
     def describe(self):
         pass
 
     @abstractmethod
     def collect(self):
-        pass
\ No newline at end of file
+        pass
+
+    def get_handles(self):
+        """
+        Return handles to all the servers from inventory.
+        :return: dict of handles
+                e.g. self.handles = {"server_name": "<server_handle>", ...}
+        """
+        server_list = self.get_inventory()
+        print("Create handles for Ucsm Servers.")
+        for server in range(len(server_list)):
+            try:
```

Comment:
remove this try-except block. It Would only be catching a login-error when running UcsmServer. In there you have try-excepted the login method already, so no need to do it again.